OutOfSchool/OutOfSchool.WebApi.Tests/Services/BlockedProviderParentServiceTests.cs

```diff
@@ -0,0 +1,228 @@
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.Tests.Common;
+using OutOfSchool.Tests.Common.TestDataGenerators;
+using OutOfSchool.WebApi.Models.BlockedProviderParent;
+using OutOfSchool.WebApi.Services;
+using OutOfSchool.WebApi.Util;
+using OutOfSchool.WebApi.Util.Mapping;
+
+namespace OutOfSchool.WebApi.Tests.Services;
+
+[TestFixture]
+public class BlockedProviderParentServiceTests
+{
+    private BlockedProviderParentService service;
+
+    private Mock<IBlockedProviderParentRepository> blockedProviderParentRepositoryMock;
+    private Mock<ILogger<BlockedProviderParentService>> loggerMock;
+    private Mock<IStringLocalizer<SharedResource>> localizerMock;
+    private IMapper mapper;
+    private Mock<INotificationService> notificationServiceMock;
+    private Mock<IParentRepository> parentRepositoryMock;
+
+    private BlockedProviderParentBlockDto blockDto;
+    private BlockedProviderParentUnblockDto unblockDto;
+    private string userId;
+    private BlockedProviderParent blockEntity;
+    private BlockedProviderParent unblockEntity;
+
+    [SetUp]
+    public void SetUp()
+    {
+        blockedProviderParentRepositoryMock = new Mock<IBlockedProviderParentRepository>();
+        loggerMock = new Mock<ILogger<BlockedProviderParentService>>();
+        localizerMock = new Mock<IStringLocalizer<SharedResource>>();
+        mapper = TestHelper.CreateMapperInstanceOfProfileTypes<CommonProfile, MappingProfile>();
+        notificationServiceMock = new Mock<INotificationService>();
+        parentRepositoryMock = new Mock<IParentRepository>();
+
+        service = new BlockedProviderParentService(
+            blockedProviderParentRepositoryMock.Object,
+            loggerMock.Object,
+            localizerMock.Object,
+            mapper,
+            notificationServiceMock.Object,
+            parentRepositoryMock.Object);
+
+        blockDto = new BlockedProviderParentBlockDto()
+        {
+            ParentId = Guid.NewGuid(),
+            ProviderId = Guid.NewGuid(),
+            Reason = "Reason to block user",
+        };
+        unblockDto = new BlockedProviderParentUnblockDto()
+        {
+            ParentId = Guid.NewGuid(),
+            ProviderId = Guid.NewGuid(),
+        };
+        userId = Guid.NewGuid().ToString();
+        blockEntity = new BlockedProviderParent()
+        {
+            Id = Guid.NewGuid(),
+            ParentId = blockDto.ParentId,
+            ProviderId = blockDto.ProviderId,
+        };
+        unblockEntity = new BlockedProviderParent()
+        {
+            Id = Guid.NewGuid(),
+            ParentId = unblockDto.ParentId,
+            ProviderId = unblockDto.ProviderId,
+        };
+    }
+
+    #region Block
+
+    [Test]
+    public async Task Block_WhenDtoAndUserIdIsValid_ShouldSendNotificationAndReturnSuccess()
```

Comment:
(Test readability) Could some if `It.IsAny` be replaced by real data? 
I expect that some simple data like `Guid`, `string`, are known and could be checked during this test. 
This also would help to set clear expectations.

Same for other tests

Reply:
Thank you! I replaced them.